internal/handlers/user.go

```diff
@@ -24,10 +24,18 @@ func (api *API) CreateUserIfNotExists() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		id := c.GetString(gin.AuthUserKey)
 		username := c.GetString(AuthUserName)
-		_, err := api.createUserIfNotExists(c.Request.Context(), id, username)
+		session, err := api.sessionManager.Start(c.Request.Context(), c.Writer, c.Request)
```

Comment:
This would only work for browser based sessions.  sessions manager creates a random key for each user and stores it in a browser cookie.  This key is what is used to get the session data out of redis.  

We need this cache site to work even for non-browser sessions.  So you will need to use redis directly as a map and not as a user session.

Reply:
@chirino, thanks for the clarification; I can update it to work for non-browser sessions.

Reply:
this article might help: https://blog.petehouston.com/use-redis-to-cache-data-in-go/